### ISSUE TITLE
How do I make `sbt scapegoat` not run `scapegoatClean`?

### ISSUE THREAD

**Issue by GordonGustafson** [sksamuel/sbt-scapegoat#39]
Currently when I run `sbt scapegoat` it recompiles and reanalyzes my entire project. I want it to only analyze what was already going to be recompiled. I added `scapegoatRunAlways := false` to my `build.sbt` because it's documented as doing exactly what I want. It had no effect, and looking at the source for this project I see that it only affects the behavior of `scapegoatCleanTask`, which can only be invoked manually.

I found this in the source:

      (compile in Scapegoat) <<= (compile in Scapegoat) dependsOn scapegoatClean,
      scapegoat := (compile in Scapegoat).value,

Since `scapegoat` depends on `scapegoatClean`, what I want seems to be impossible. The top line were changed to use `scapegoatCleanTask` (which respects `scapegoatRunAlways`) instead of `scapegoatClean`, all would be well.

            (compile in Scapegoat) <<= (compile in Scapegoat) dependsOn scapegoatCleanTask,

**Comment by sksamuel** [comment]
Yes it does clean on each run. That was part of the design so you get a full picture. I guess the devils advocate argument against your question is that if you only evaluate incrementally, it will lose the previous "errors"

**Comment by GordonGustafson** [comment]
Cleaning *by default* on each run is fine. I just want to be able to configure it to not clean on each run. If the impotency of `scapegoatRunAlways` isn't considered a bug, is there any way I can achieve it's intended purpose of not cleaning on each run?

**Comment by sksamuel** [comment]
It can be changed to support what you want.

**Comment by dskrvk** [comment]
Related: https://github.com/sksamuel/sbt-scapegoat/issues/31
